klayout_package/python/scripts/simulations/elmer/paraview_macro.py

```diff
@@ -0,0 +1,153 @@
+            threshold_display = Show(threshold, render_view, "UnstructuredGridRepresentation")
+            threshold_display.Representation = "Surface"
+            threshold_display.SetScalarBarVisibility(render_view, True)
+
+            if i + 1 < len(registration_names):
+                Hide(threshold, render_view)
+
+
+def render_data(simulation_folder_path: str, simulation_name: str, cross_section: bool, render_view: object):
+
+    layout = GetLayout()
+    layout.SetSize(720, 650)
+
+    new_box = []
+    if simulation_name:
+        try:
+            path_to_simulation_json = f"{simulation_folder_path}/{simulation_name}.json"
+            with open(path_to_simulation_json, "r") as f:
+                data = json.load(f)
+            current_box = GetActiveSource().GetDataInformation().GetBounds()
+            try:
+                new_box = {
+                    "x_min": data["box"]["p1"]["x"] / 1e6,
+                    "y_min": data["box"]["p1"]["y"] / 1e6,
+                    "x_max": data["box"]["p2"]["x"] / 1e6,
+                    "y_max": data["box"]["p2"]["y"] / 1e6,
+                    "z_min": current_box[4],
+                    "z_max": current_box[5],
+                }
+            except KeyError as e:
+                new_box = {
+                    "x_min": data["parameters"]["source_box"]["p1"]["x"] / 1e6,
+                    "y_min": data["parameters"]["source_box"]["p1"]["y"] / 1e6,
+                    "x_max": data["parameters"]["source_box"]["p2"]["x"] / 1e6,
+                    "y_max": data["parameters"]["source_box"]["p2"]["y"] / 1e6,
+                    "z_min": current_box[4],
+                    "z_max": current_box[5],
+                }
+        except (FileNotFoundError, ValueError, KeyError) as e:
+            print("Failed to zoom automatically. Not a critical error.")
+
+    render_view.ResetActiveCameraToNegativeZ()
+    if new_box:
+        if not cross_section:
+            render_view.ResetCamera(
+                new_box["x_min"],
+                new_box["x_max"],
+                new_box["y_min"],
+                new_box["y_max"],
+                new_box["z_min"],
+                new_box["z_max"],
+                True,
+                0.9,
+            )
+        else:
+            render_view.ResetCamera(True, 0.8)
+            render_view.ResetCamera(True, (new_box["y_max"] / new_box["x_max"]) / 2)
+            render_view.CameraFocalPoint = [3 * new_box["x_max"], 0, 0]
+    else:
+        render_view.ResetCamera(True, 0.9)
+
+
+def run_macro(data_files: list[str], sif_files: list[str], cross_section: bool):
+    """Calls other functions to run applicable parts of a ParaView visualisation."""
+
+    paraview.simple._DisableFirstRenderCameraReset()
+    render_view = GetActiveViewOrCreate("RenderView")
+
+    simulation_folder_path = str(Path.cwd())
+    registration_names = []
+    simulation_name = vtu_pvtu = new_box = None
```

Comment:
```suggestion
    simulation_name, vtu_pvtu = None, None
```
new_box I think not used here?

Reply:
Done. ...